modules/sdk-lib-mpc/src/tss/ecdsa-dkls/util.ts

```diff
@@ -27,3 +31,34 @@ export function combinePartialSignatures(round4MessagePayloads: Uint8Array[], rH
     S: new Uint8Array(bigIntToBufferBE(normalizedSig.s)),
   };
 }
+
+/**
+ * Verify a DKLs Signature and serialize it to recid:r:s:publickey format.
+ * @param message - message that was signed.
+ * @param dklsSignature - R and S values of the ECDSA signature.
+ * @param commonKeychain - public key appended to chaincode in hex.
+ * @param hash - optional hash function to apply on message before verifying. Default is sha256.
+ * @param shouldHash - flag to determine whether message should be hashed before verifying.
+ * @returns {string} - serialized signature in `recid:r:s:publickey` format
+ */
+export function verifyAndConvertDklsSignature(
+  message: Buffer,
+  dklsSignature: DeserializedDklsSignature,
+  commonKeychain: string,
+  hash?: Hash,
+  shouldHash = true
+): string {
+  const messageToVerify = shouldHash ? (hash || createHash('sha256')).update(message).digest() : message;
+  const pub0 = secp256k1.ecdsaRecover(Buffer.concat([dklsSignature.R, dklsSignature.S]), 0, messageToVerify, true);
+  const pub1 = secp256k1.ecdsaRecover(Buffer.concat([dklsSignature.R, dklsSignature.S]), 1, messageToVerify, true);
+  const truePub = commonKeychain.slice(0, 66);
+  let recId = 0;
+  if (truePub === Buffer.from(pub1).toString('hex')) {
+    recId = 1;
+  } else if (truePub !== Buffer.from(pub0).toString('hex')) {
+    throw Error('Invalid Signature');
+  }
+  return `${recId}${delimeter}${Buffer.from(dklsSignature.R).toString('hex')}${delimeter}${Buffer.from(
+    dklsSignature.S
+  ).toString('hex')}${delimeter}${commonKeychain.slice(0, 66)}`;
```

Comment:
could use `truePub` in place of `commonKeychain.slice(0, 66)`

Reply:
oh true

Reply:
fixed